drivers/crypto/crypto_intel_sha.c

```diff
@@ -0,0 +1,335 @@
+		len /= sizeof(uint32_t);
+		uint32_t counter = 0;
+		/* The index of a last element in the sha result buffer. */
+		uint32_t last_idx = (sizeof(self->dfsha->sha_result) / sizeof(uint32_t)) - 1;
+
+		for (counter = 0; counter != len; counter++) {
+			((uint32_t *)dst)[counter] =
+				((uint32_t *)self->dfsha->sha_result)[last_idx - counter];
+		}
+
+		err = 0;
+	}
+
+	return err;
+}
+
+static int intel_sha_device_get_hash(struct device *dev, void *buf_out, size_t buf_out_size)
+{
+	int err;
+	struct sha_container *const self = dev->data;
+
+	if (dev == NULL)
+		return -ENODEV;
+	if (buf_out == NULL)
+		return -EINVAL;
+	/* wait until not busy */
+	while (self->dfsha->shasts.part.busy) {
+	}
+
+	err = intel_sha_copy_hash(self, buf_out, buf_out_size);
+
+	return err;
+}
+
+static int intel_sha_compute(struct hash_ctx *ctx, struct hash_pkt *pkt, bool finish)
+{
+	xthal_dcache_region_writeback((uint8_t *)(pkt->in_buf), pkt->in_len);
+
+	int ret;
+	struct sha_container *self = (struct sha_container *const)(ctx->device)->data;
+	struct sha_session *session = (struct sha_session *)ctx->drv_sessn_state;
+
+	/* set algo */
+	self->dfsha->shactl.full = 0x0;
+	self->dfsha->shactl.part.algo = session->algo;
+
+	/* restore ctx */
+	self->dfsha->shaaldw0 = session->sha_ctx.shaaldw0;
+	self->dfsha->shaaldw1 = session->sha_ctx.shaaldw1;
+
+	ret = intel_sha_regs_cpy((void *)self->dfsha->initial_vector,
+				 (void *)session->sha_ctx.initial_vector,
+				 sizeof(self->dfsha->initial_vector));
+	if (ret)
+		return ret;
+
+	ret = intel_sha_regs_cpy((void *)self->dfsha->sha_result,
+				 (void *)session->sha_ctx.sha_result,
+				 sizeof(self->dfsha->sha_result));
+
+	if (ret)
+		return ret;
+
+	/* compute hash */
+	do {
+		size_t frag_length = pkt->in_len > SHA_API_MAX_FRAG_LEN ?
+					     SHA_API_MAX_FRAG_LEN :
+						   pkt->in_len;
+
+		if ((frag_length == pkt->in_len) && finish) {
+			session->state_.part.state = SHA_LAST;
+		}
+
+		ret = intel_sha_device_run(ctx->device, pkt->in_buf, frag_length, frag_length,
+					   session->state_.full);
+		if (ret != 0) {
+			return ret;
+		}
+
+		/* set state for next iteration */
+		session->state_.part.hrsm = SHA_HRSM_ENABLE;
+		session->state_.part.state = SHA_MIDLE;
+
+		pkt->in_len -= frag_length;
+		pkt->in_buf += frag_length;
+	} while (pkt->in_len > 0);
+
+	if (finish) {
+		volatile size_t output_size;
+		struct sha_container *self =
+			(struct sha_container *const)(ctx->device)->data;
+		switch (self->dfsha->shactl.part.algo) {
+		case CRYPTO_HASH_ALGO_SHA224:
+			output_size = SHA224_ALGORITHM_HASH_SIZEOF;
+			break;
+		case CRYPTO_HASH_ALGO_SHA256:
+			output_size = SHA256_ALGORITHM_HASH_SIZEOF;
+			break;
+		case CRYPTO_HASH_ALGO_SHA384:
+			output_size = SHA384_ALGORITHM_HASH_SIZEOF;
+			break;
+		case CRYPTO_HASH_ALGO_SHA512:
+			output_size = SHA512_ALGORITHM_HASH_SIZEOF;
+			break;
+		default:
+			return -ENOTSUP;
+		}
+		ret = intel_sha_device_get_hash(ctx->device, pkt->out_buf, output_size);
```

Comment:
Missing error handling?

Is the execution of
https://github.com/zephyrproject-rtos/zephyr/blob/18cfe39d31619ee58d0b8e74c37bdd04f1ee9d36/drivers/crypto/crypto_intel_sha.c#L265-L270
desired when `ret` is `< 0`?

Reply:
If no/invalid output is produced it just reorganizes the output array and returns error code, so no impact there.
Added error check for code clarity anyway.